src/components/timers/XY.tsx

```diff
@@ -1,3 +1,122 @@
-const XY = () => null;
+import React, { useState, useEffect } from "react";
+
+type TimerProps = {
+  timePerRound: number;
+  rounds: number;
+};
+// timePerRound(30);
+// rounds(5);
+
+const XY: React.FC<TimerProps> = ({ timePerRound, rounds }) => {
+  const [round, setRound] = useState<number>(1);
+  const [count, setCount] = useState<number>(timePerRound);
+  const [isRunning, setIsRunning] = useState<boolean>(false);
+  const [message, setMessage] = useState<string | null>(null);
+  const [isPaused, setIsPaused] = useState<boolean>(false);
+
+  useEffect(() => {
+    // set the interval to 0
+    let intervalId: number;
+// check if the timer is running
+    if (isRunning) {
+      intervalId = setInterval(() => {
+        setCount((prev) => {
+            // if the count is greater than 0, decrease the count by 1 second
+          if (prev > 0) {
+            return prev - 1;
+            // otherwise, check if the round is less than the total rounds
+          } else if (round < rounds) {
+            setRound((prev) => prev + 1);
+            return timePerRound;
+            // otherwise, end the timer
+          } else {
+            setMessage("Finished");
+            setIsRunning(false);
+            return 0;
+          }
+        });
+        // run the interval every 1 second
+      }, 1000);
+    }
+// clear the interval
+    return () => clearInterval(intervalId);
+    // run the effect whenever the timer is running, the round changes, the time per round changes, or the rounds change
+  }, [isRunning, round, timePerRound, rounds]);
+
+  // start the timer function
+  const startTimer = () => {
```

Comment:
When I start the timer at the end of the rounds, it resets to the start of the last round. It should reset to the beginning of round 1.